OpenIPSL/Electrical/Renewables/PSSE/WindDriveTrain/WTDTA1.mo

```diff
@@ -0,0 +1,185 @@
+  Modelica.Blocks.Sources.RealExpression realExpression(y=add.y) annotation (
+      Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=180,
+        origin={-30,20})));
+  Modelica.Blocks.Math.Add add1(k1=-1)
+    annotation (Placement(transformation(extent={{-48,-70},{-28,-50}})));
+  Modelica.Blocks.Math.Gain w0__(k=w0) annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={60,0})));
+  Modelica.Blocks.Math.Add add2(k1=-1)
+    annotation (Placement(transformation(extent={{24,-84},{44,-104}})));
+  Modelica.Blocks.Math.Add3 add3_2(k1=+1,
+    k2=-1,
+    k3=+1)
+    annotation (Placement(transformation(extent={{-80,-76},{-60,-56}})));
+  Modelica.Blocks.Sources.RealExpression realExpression1(y=integrator2.y)
+                                                                 annotation (
+      Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=180,
+        origin={-70,84})));
+  Modelica.Blocks.Continuous.Integrator integrator2(
+    k=Kshaft,
+    initType=Modelica.Blocks.Types.Init.InitialState,
+    y_start=T0)
+    annotation (Placement(transformation(extent={{80,-10},{100,10}})));
+  Modelica.Blocks.Sources.RealExpression realExpression2(y=integrator2.y)
+                                                                 annotation (
+      Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=180,
+        origin={-70,-84})));
+  Modelica.Blocks.Math.Gain w0_(k=w0) annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={64,-94})));
+  Modelica.Blocks.Continuous.Integrator integrator3(k=1, y_start=0)
+    annotation (Placement(transformation(extent={{84,-104},{104,-84}})));
+
+  Modelica.Blocks.Math.Division division
+    annotation (Placement(transformation(extent={{-110,76},{-90,56}})));
+  Modelica.Blocks.Math.Add add3(k2=+1)
+    annotation (Placement(transformation(extent={{20,70},{40,90}})));
+  Modelica.Blocks.Sources.Constant const(k=1)
+    annotation (Placement(transformation(extent={{-20,76},{0,96}})));
+  Modelica.Blocks.Math.Add add4(k2=+1)
+    annotation (Placement(transformation(extent={{64,-56},{84,-36}})));
+  Modelica.Blocks.Sources.Constant const1(k=1)
+    annotation (Placement(transformation(extent={{24,-50},{44,-30}})));
+  Modelica.Blocks.Math.Division division1
+    annotation (Placement(transformation(extent={{-112,-76},{-92,-56}})));
+protected
+  parameter OpenIPSL.Types.PerUnit Ht = H*Htfrac;
+  parameter OpenIPSL.Types.PerUnit Hg = H - Ht;
+  parameter OpenIPSL.Types.PerUnit Kshaft = 2*Ht*Hg*(2*pi*Freq1)^2/(H*w0);
+  parameter Modelica.Units.SI.AngularVelocity w0 = 2*pi*fn;
+  parameter OpenIPSL.Types.PerUnit p00(fixed=false);
+  parameter OpenIPSL.Types.PerUnit W0(fixed=false);
+  parameter OpenIPSL.Types.PerUnit T0 = p00;
+
+initial equation
+
+   p00 = P0;
+   W0 = W_0;
```

Comment:
`p00` and `W0` are not needed here. You could simply connect W_0 on the diagram and make `T0` fixed= false and so set it to `T0 = P0` in the initial equation. That way you got rid of 2 intermediate parameters.